olive/passes/onnx/transformer_optimization.py

```diff
@@ -94,30 +94,62 @@ def sd_model_types():
         return ("unet", "vae", "clip")
 
     @staticmethod
-    def _set_sd_fusion_options(run_config: Dict[str, Any], use_float16: bool):
+    def _set_sd_fusion_options(run_config: Dict[str, Any], pass_config: Dict[str, Any]):
         """Configures fusion options for stable diffusion models"""
         import onnxruntime as ort
 
-        input_model_type = run_config["model_type"]
+        ort_version = version.parse(ort.__version__)
+        is_ort_1_13_or_older = ort_version < version.parse("1.14.0")
+        # is_ort_1_14 = ort_version >= version.parse("1.14.0") and ort_version < version.parse("1.15.0")
 
         # default to no specific fusion options in earlier releases of ORT
-        if version.parse(ort.__version__) < version.parse("1.14.0"):
+        if is_ort_1_13_or_older:
             return
 
-        from onnxruntime.transformers.fusion_options import FusionOptions
+        is_ort_1_15_0_or_newer = ort_version >= version.parse("1.15.0")
+        is_ort_1_15_1_or_newer = ort_version >= version.parse("1.15.1")
 
-        is_ort_115_or_newer = version.parse(ort.__version__) >= version.parse("1.15.0")
-        if not is_ort_115_or_newer and input_model_type != "unet":
+        input_model_type = run_config["model_type"]
+        if not is_ort_1_15_0_or_newer and input_model_type != "unet":
             # 'vae' and 'clip' are only recognized in ORT v1.15+. earlier versions of ORT stable diffusion
             # optimization simply use "unet" for these model types.
             run_config["model_type"] = "unet"
 
+        from onnxruntime.transformers.fusion_options import FusionOptions
+
         fusion_options = FusionOptions(run_config["model_type"])
 
-        if input_model_type == "unet":
-            fusion_options.enable_packed_kv = use_float16
-            fusion_options.enable_packed_qkv = use_float16 and is_ort_115_or_newer
-            fusion_options.enable_bias_add = is_ort_115_or_newer
+        # TODO: remove dml_future when ORT 1.15.1 with future version of DML is released
+        # This "provider" value is simply a way to test in-development changes without having
+        # to bump the ORT version.
+        dml_future = pass_config["target_provider"] == "directml_future"
+
+        if pass_config["target_provider"] == "directml" or dml_future:
+            # Some of these fusions are disabled because they provide no performance advantage,
+            # and it's preferable to limit ops outside the ONNX domain.
+            fusion_options.enable_gelu = is_ort_1_15_0_or_newer
+            fusion_options.enable_layer_norm = is_ort_1_15_0_or_newer
+            fusion_options.enable_attention = is_ort_1_15_1_or_newer or dml_future
+            fusion_options.use_multi_head_attention = is_ort_1_15_1_or_newer or dml_future
+            fusion_options.enable_skip_layer_norm = is_ort_1_15_0_or_newer
```

Comment:
Propose disabling since it buys no advantage for DML (per conversation with Pat/Justin) and just complicates mapping to WebNN.

Reply:
Unless Pat says otherwise, I'd apply the same to `BiasSkipLayerNorm` too. I don't even see a registration for it in OperatorRegistration.

Reply:
For `EmbedNormalization` (nice ASCII art in the ORT EP btw), I wonder if this is another case where *if* the model already has one, then we need a functional decomposition for it (just so ORT doesn't fall back to CPU because CPU registered the contrib op even though DML was registered as a higher priority EP), but otherwise we don't really get benefit from it 🤔...